proj/src/main/webapp/script.js

```diff
@@ -37,12 +39,23 @@ if(btnSignUp){
         auth.useDeviceLanguage();
 
         const promise = auth.createUserWithEmailAndPassword(emailVal, passVal).then(function(){
-            auth.currentUser.updateProfile({displayName: fname.value + " " + lname.value}).then(function(){
+            const user = auth.currentUser;
+            user.updateProfile({
+                displayName: fname.value + " " + lname.value
+                }).then(function(){
                 console.log("display name updated successfully");
-                window.location.replace("chat.html");
+                if(fname && lname && tagStr){
```

Comment:
Does this mean we don't want to store tagStr if they didn't enter a last name for example? Or that we don't want to store their name if they didn't enter any tags? Maybe you can just set the fields that are present, or if all is required, i'd do some kind of validation before creating the user.

We previously set their display name using their fname and lname so I would think up there we might want to also check if fname and lname is populated.